hub/api/dataset.py

```diff
@@ -0,0 +1,35 @@
+from hub.util.slice import merge_slices
+
+
+class Dataset:
+    def __init__(self, path: str, mode: str = "a", ds_slice: slice = slice(None)):
+        """
+        Initialize a new or existing dataset.
+
+        Parameters
+        ----------
```

Comment:
i know this is just a draft, but can you please use [google docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)?

Reply:
This is exactly the reason I put up the draft! More nitpicks please ;)